src/BugsnagUnity/FileManager.cs

```diff
@@ -106,30 +128,64 @@ internal static void CacheSession(string reportId)
             }
         }
 
+        internal static void CacheEvent(string reportId)
+        {
+            var eventReport = GetPendingEventReport(reportId);
+            if (eventReport == null)
+            {
+                return;
+            }
+            using (var stream = new MemoryStream())
+            using (var reader = new StreamReader(stream))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { AutoFlush = false })
+            {
+                SimpleJson.SerializeObject(eventReport.GetSerialisableEventReport(), writer);
+                writer.Flush();
+                stream.Position = 0;
+                var jsonString = reader.ReadToEnd();
+                var path = EventsDirectory + "/" + eventReport.Id + ".event";
+                WriteToDisk(jsonString, path);
+                CheckForMaxCachedEvents();
+            }
+        }
+
+        private static void CheckForMaxCachedEvents()
+        {
+            var filesCount = _cachedEvents.Length;
+            while (filesCount > _configuration.MaxPersistedEvents)
+            {
+                RemoveOldestFile(_cachedEvents);
+                filesCount = _cachedEvents.Length;
+            }
+        }
+
         private static void CheckForMaxCachedSessions()
         {
             var filesCount = _cachedSessions.Length;
             while(filesCount > _configuration.MaxPersistedSessions)
             {
-                RemoveOldestSession();
+                RemoveOldestFile(_cachedSessions);
                 filesCount = _cachedSessions.Length;
             }
         }
 
-        private static void RemoveOldestSession()
+        private static void RemoveOldestFile(string[] filePaths)
```

Comment:
It doesn't look like we actually remove the entry from `_cachedSessions` or `_cachedEvents` after deletion.  Is this important?  Is it possible that this could be called twice without them being repopulated?

Reply:
They are repopulated whenever accessed: 
`        private static string[] _cachedSessions => Directory.GetFiles(SessionsDirectory, "*.session");
`

Reply:
oh I got confused by the `_` prefix.  That's normally used for fields rather than properties.

Reply:
PascalCase would be usual for a property.